### PR TITLE
[go-gen] Add generation of router, jsonMiddleware and empty handlers to main.go

### PR DESCRIPTION
**Pull request by willjvsmith**
* Adds generation of router, jsonMiddleware functions
* Adds empty handler declarations to make the integration tests pass
* Updates tests

Next up is the main functions

### CODE REVIEW

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainGenerator.scala [hunk 1]
```diff
@@ -37,15 +38,46 @@ object GoServiceMainGenerator {
...
         //s"valid ${doubleQuote("github.com/asaskevich/govalidator")}",
         doubleQuote("github.com/google/uuid"),
-        //doubleQuote("github.com/gorilla/mux"),
+        doubleQuote("github.com/gorilla/mux"),
+      ),
+    )
+
+  private[service] def generateRouter(root: ServiceRoot, operations: Set[CRUD]): String = {
+    val handleFuncs =
+      for (operation <- operations.toSeq.sorted)
```
Comment (Lewiky) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainGenerator.scala:53]: What do enums sort by?

**Review by Lewiky — APPROVED** [review]
Looks good

File: src/test/scala/temple/testfiles/simple.temple [hunk 2]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (jaylees14) [inline src/test/scala/temple/testfiles/simple.temple:1]: Now you're just trying to increase your lines changed 😉

**Review by jaylees14 — APPROVED** [review]
Nice stuff, easy to follow :)

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainGenerator.scala:53]: Some @xsanda magic

File: src/test/scala/temple/testfiles/simple.temple (hunk 2 above)
Comment (willjvsmith) [reply src/test/scala/temple/testfiles/simple.temple:1]: 😉

**Pull request merged**